internal/common/common.go

```diff
@@ -33,17 +49,42 @@ func CheckForVersionUpdate() {
 }
 
 // versionCheck compares local to remote version
-func versionCheck() (res *latest.CheckResponse, skip bool) {
-	githubTag := &latest.GithubTag{
-		Owner:             "kubefirst",
-		Repository:        "kubefirst",
-		FixVersionStrFunc: latest.DeleteFrontV(),
-	}
-	res, err := latest.Check(githubTag, strings.Replace(configs.K1Version, "v", "", 1))
+func versionCheck() (res *CheckResponse, skip bool) {
+	var latestVersion string
+
+	resp, err := http.Get("https://raw.githubusercontent.com/Homebrew/homebrew-core/master/Formula/k/kubefirst.rb")
+
 	if err != nil {
 		fmt.Printf("checking for a newer version failed with: %s", err)
 		return nil, true
 	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusOK {
+		bodyBytes, err := io.ReadAll(resp.Body)
+
+		if err != nil {
+			fmt.Printf("checking for a newer version failed with: %s", err)
+			return nil, true
+		}
+
+		bodyString := string(bodyBytes)
+		for _, sentence := range strings.Split(bodyString, "\n") {
+			if strings.Contains(sentence, "url \"https://github.com/kubefirst/kubefirst/archive/refs/tags/") {
+				re := regexp.MustCompile(`.*/v(.*).tar.gz"`)
+				matches := re.FindStringSubmatch(sentence)
+				latestVersion = matches[1]
+			}
+		}
```

Comment:
I would probably skip this and regex the whole body string. If I had no matches, then throw an error

Reply:
Good call!